File: src/java/src/main/java/com/scmworx/RiskPrediction.java
package com.scmworx;

import java.util.List;

public class RiskPrediction {
    private final SupplyChainModel model;

    public RiskPrediction(SupplyChainModel model) {
        this.model = model;
    }

    public void predictRisk(List<double[]> features) {
        double[] predictions = model.predict(features);
        for (int i = 0; i < predictions.length; i++) {
            System.out.println("Prediction for instance " + i + ": " + (predictions[i] > 0.5 ? "High Risk" : "Low Risk"));
        }
    }
}

File: src/csharp/SCMWorx/SCMWorx/RiskPrediction.cs
using System;
using System.Collections.Generic;

namespace SCMWorx
{
    public class RiskPrediction
    {
        private readonly SupplyChainModel model;

        public RiskPrediction(SupplyChainModel model)
        {
            this.model = model;
        }

        public void PredictRisk(List<double[]> features)
        {
            double[] predictions = model.Predict(features);
            for (int i = 0; i < predictions.Length; i++)
            {
                Console.WriteLine("Prediction for instance " + i + ": " + (predictions[i] > 0.5 ? "High Risk" : "Low Risk"));
            }
        }
    }
}
